panoramapublic/src/org/labkey/panoramapublic/pipeline/CopyExperimentFinalTask.java

```diff
@@ -524,11 +536,41 @@ private void updatePermissions(User formSubmitter, ExperimentAnnotations targetE
         SecurityPolicyManager.savePolicy(newPolicy);
     }
 
-    private void assignPanoramaPublicSubmitterRole(User user, MutableSecurityPolicy policy)
+    private void assignPanoramaPublicSubmitterRole(MutableSecurityPolicy policy, Logger log, User... users)
     {
-        if (user != null)
-        {
+        Arrays.stream(users).filter(Objects::nonNull).forEach(user -> {
+            log.info("Assigning " + PanoramaPublicSubmitterRole.class.getSimpleName() + " to " + user.getEmail());
             policy.addRoleAssignment(user, PanoramaPublicSubmitterRole.class, false);
+        });
+    }
+
+    private void addToSubmittersGroup(Container project, Logger log, User... users)
+    {
+        Arrays.stream(users).filter(Objects::nonNull).forEach(user -> {
+            // This group already exists in the Panorama Public project on panoramaweb.org.
+            // CONSIDER: Make this configurable through the Panorama Public admin console.
+            addToGroup(user, "Panorama Public Submitters", project, log);
+        });
+    }
+
+    private void addToGroup(User user, String groupName, Container project, Logger log)
+    {
+        Integer groupId = SecurityManager.getGroupId(project, groupName, false);
+        if (groupId != null)
```

Comment:
Would it be useful to log if we can't find the group? Does the test create it today?

Reply:
Done.